get_goes_x.py

```diff
@@ -11,99 +11,148 @@
 """
 
 import sys
-import urllib.request, urllib.error, urllib.parse
+import urllib.request
+import urllib.error
+import urllib.parse
 import argparse
 import tables
 import time
 import json
 
 import numpy as np
-from astropy.io import ascii
 from astropy.table import Table, join
 from astropy.time import Time
 
-parser = argparse.ArgumentParser(description='Get GOES_X data')
-parser.add_argument('--h5',
-                    default='GOES_X.h5',
-                    help='HDF5 file name')
-parser.add_argument('--satellite',
-                    default=16,
-                    type=int,
-                    help='Select which satelite from the json file by int id')
-args = parser.parse_args()
+# URLs for 6 hour and 7 day JSON files
+URL_6H = 'https://services.swpc.noaa.gov/json/goes/primary/xrays-6-hour.json'
+URL_7D = 'https://services.swpc.noaa.gov/json/goes/primary/xrays-7-day.json'
+
+
+def get_options():
+    parser = argparse.ArgumentParser(description='Get GOES_X data')
+    parser.add_argument('--h5',
+                        default='GOES_X.h5',
+                        help='HDF5 file name')
+    parser.add_argument('--satellite',
```

Comment:
Remind me why are we using a program-level specification of satellite to filter the realtime data?  Looks like I didn't notice this the first time round, but I don't fully understand why this is helpful. At the least I would make the default `None` and only filter if it is specifically provided.

Reply:
https://github.com/sot/arc/pull/44#discussion_r398571457 

If you wanted it cmdline configureable I thought it had to be used as a filter and I also figured (that since I didn't see a reason for the cmdline) that having a default was also reasonable to simplify the default task schedule.

Reply:
It you think there's a chance we "won't care" if they *add* another satellite... that seemed risky to me.

Reply:
I guess the original code when I wrote my comment probably had satellite hardwired. It seems that the way it is written the code will (functionally) fail when they change the primary satellite. In other words, if they put up GOES-17 and make it prime, then our code will stop adding new data until we update the cron job.  Updating the cron job would need to be synchronized with the SWPC change to not lose any data.

I can't see any danger to passing through whatever X-ray data SWPC calls prime as long as the satellite number is captured, but maybe I'm missing something?

Reply:
I think we're both just guessing what data might look like if the change or add a satellite.  When you say "whatever X-ray data SWPC calls prime" ... I don't even know if they would do that as they move forward compared to just throwing more data in the file.  I think the process section might need reorganization to keep one row per satellite in that case and I don't see that quite worth the work (without a clear idea about what data would be or what to test)?

Reply:
I'm saying "there is a good chance things will just work when they swap to a new prime satellite".  (I would guess they would not put multiple satellite data in one file from past experience). You are concerned things might break.  But let's be success oriented and then fix if things break. This is a situation where having the code break is not a real problem, and a pending change certainly won't happen without plenty of advance notice.

You can split the difference by keeping the command line arg but making the default None and *not* providing `--satellite` in the task schedule.

Reply:
I'm not sure which of our strategies is actually more success-oriented but that's fine.